yaksh/templates/yaksh/showquestions.html

```diff
@@ -38,18 +38,55 @@ <h5 style="padding-left: 20px;">Filters</h5>
     <div class="col-md-3">
         {{ form.marks }}
     </div>
-</div>
-<br>
-    <button class="btn btn-primary" type="button" onClick='location.replace("{{URL_ROOT}}");'>Clear Filters</button>
 <br>
+<h4 style="padding-left: 20px;">Or</h4>
+
+<h4 style="padding-left: 20px;">Search using Tags: </h4>
+</div>
+<!-- Searching Tags -->
+{% csrf_token %}
+    <div class="col-md-14">
+    <div class="input-group">
+    <span class="input-group-addon" id="basic-addon1">Search Questions </span>
+    <input type="text" id="question_tags" name="question_tags" class="form-control"
+     placeholder="Search using comma separated Tags">
+    <span class="input-group-btn">
+    <button class="btn btn-default" type="submit">Search</button>
+    </span>
+    <div class="col-md-6">
+    <select class="form-control" id="sel1" onchange="append_tag(this);">
+    {% if all_tags %}
+     <option value="" disabled selected>Available Tags</option>
+    {% for tag in all_tags %}
+    <option>
+        {{tag}}
+    </option>
+    {% endfor %}
+    {% else %}
+     <option value="" disabled selected>No Available Tags</option>
+    {% endif %}
+    </select>
+    </div>
+    </div>
+     </div>
+<br><br>
+<button class="btn btn-primary" type="button" onClick='location.replace("{{URL_ROOT}}");'>
+ Clear Filters</button>
 
 <div id="filtered-questions">
+{% if not search_result %}
```

Comment:
if no such tags then user will not know! It will display all questions, giving an impression that all questions belong to the searched tags.